src/core/flow.js

```diff
@@ -278,7 +278,8 @@ export function FlowWithPhases({
   const startTurn = function(state, config) {
     const ctx = { ...state.ctx };
     const G = config.onTurnBegin(state.G, ctx);
-    const _undo = [{ G, ctx }];
+    const { random, ...ctxWithoutAPI } = ctx; // eslint-disable-line no-unused-vars
```

Comment:
I'd like to hide how `random` is attached to `ctx`. How about a function inside _random.js_ that performs that action?

How does this scale to, say, the Events API? Does this have to be attached/stripped likewise?

Reply:
Done. The Events API will probably have a similar `attach / detach` mechanism for ctx.